pkg/kp/kv.go

```diff
@@ -240,6 +241,35 @@ func (c consulStore) ListPods(keyPrefix string) ([]ManifestResult, time.Duration
 	return ret, queryMeta.RequestTime, nil
 }
 
+// WatchPod is like WatchPods, but for a single key only. The output channel
+// may contain nil manifests, if the target key does not exist.
+func (c consulStore) WatchPod(key string, quitChan <-chan struct{}, errChan chan<- error, podChan chan<- ManifestResult) {
+	defer close(podChan)
+
+	kvpChan := make(chan *api.KVPair)
+	go consulutil.WatchSingle(key, c.client.KV(), kvpChan, quitChan, errChan)
+	for pair := range kvpChan {
```

Comment:
see below - the termination condition is that when quit closes, WatchSingle gets the quit and closes kvpChan, which then terminates the range loop. No need to select on quit ourselves.